src/view-utilities.js

```diff
@@ -153,29 +154,79 @@ var viewUtilities = function (cy, options) {
       shiftKeyDown = false;
     }, "keyup");
 
-    var p_start_x, p_start_y, p_end_x, p_end_y;
+    var rect_start_pos_x, rect_start_pos_y, rect_end_pos_x, rect_end_pos_y;
 
-    cy.one('tapstart', function( event){ 
+    cy.one('tapstart', function(event){ 
       if( shiftKeyDown == true){
-        p_start_x = event.position.x;
-        p_start_y = event.position.y;
+        rect_start_pos_x = event.position.x;
+        rect_start_pos_y = event.position.y;
+
       }
     });
 
-    cy.one('tapend', function( event){
-      p_end_x = event.position.x;
-      p_end_y = event.position.y;
-      var zoomLevel = Math.min( cy.width()/ ( Math.abs(p_end_x- p_start_x)), cy.height() / Math.abs( p_end_y - p_start_y));
+    cy.one('tapend', function(event){
+      rect_end_pos_x = event.position.x;
+      rect_end_pos_y = event.position.y;
+      //Find top left of rectangle
+      if( rect_start_pos_x > rect_start_pos_x){
+        var temp = rect_start_pos_x;
+        rect_start_pos_x = rect_end_pos_x;
+        rect_end_pos_x = temp;
+      }
+      if( rect_start_pos_y > rect_end_pos_y){
+        var temp = rect_start_pos_y;
+        rect_start_pos_y = rect_end_pos_y;
+        rect_end_pos_y = temp;
+      }
```

Comment:
I think the organization of the tapend after this point should be as folllows:

1. Check whether any side of the rectangle is less than 100. For example, if rect_start_pos_x is 40 and rect_end_pos_x is 100, then the difference is 60, so we need to expand the rectangle both from left and right side to make the distance 100. To achieve that, make the value of rect_start_pos_x 20 and the value of rect_end_pos_x 120. Similarly, if rect_start_pos_y is 210 and rect_end_pos_y is 300, make the value of rect_start_pos_y 205 and value of rect_end_pos_y 305. Do not use zoomLevel here.
At the end of this step, we will have a rectangle whose width and height values are at least 100. 
2. Check whether the resulting rectangle after step 1 has an intersection with the boundingBox of the graph. If there is no intersection, ignore marquee zoom.
3. Calculate the zoom level according to the rectangle resulting after step 1. 
4. Do the calculations to find the pan position. 
5. Call the animate function.